core/trino-main/src/main/java/io/trino/metadata/ResolvedFunction.java

```diff
@@ -27,122 +26,37 @@
 import io.trino.spi.type.TypeSignature;
 
 import java.util.Map;
-import java.util.Objects;
 import java.util.Set;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
-public class ResolvedFunction
+public record ResolvedFunction(
+        BoundSignature signature,
+        CatalogHandle catalogHandle,
+        FunctionId functionId,
+        FunctionKind functionKind,
+        boolean deterministic,
+        FunctionNullability functionNullability,
+        Map<TypeSignature, Type> typeDependencies,
+        Set<ResolvedFunction> functionDependencies)
 {
-    private final BoundSignature signature;
-    private final CatalogHandle catalogHandle;
-    private final FunctionId functionId;
-    private final FunctionKind functionKind;
-    private final boolean deterministic;
-    private final FunctionNullability functionNullability;
-    private final Map<TypeSignature, Type> typeDependencies;
-    private final Set<ResolvedFunction> functionDependencies;
-
-    @JsonCreator
-    public ResolvedFunction(
-            @JsonProperty("signature") BoundSignature signature,
-            @JsonProperty("catalogHandle") CatalogHandle catalogHandle,
-            @JsonProperty("id") FunctionId functionId,
-            @JsonProperty("functionKind") FunctionKind functionKind,
-            @JsonProperty("deterministic") boolean deterministic,
-            @JsonProperty("functionNullability") FunctionNullability functionNullability,
-            @JsonProperty("typeDependencies") Map<TypeSignature, Type> typeDependencies,
-            @JsonProperty("functionDependencies") Set<ResolvedFunction> functionDependencies)
+    public ResolvedFunction
     {
-        this.signature = requireNonNull(signature, "signature is null");
-        this.catalogHandle = requireNonNull(catalogHandle, "catalogHandle is null");
-        this.functionId = requireNonNull(functionId, "functionId is null");
-        this.functionKind = requireNonNull(functionKind, "functionKind is null");
-        this.deterministic = deterministic;
-        this.functionNullability = requireNonNull(functionNullability, "functionNullability is null");
-        this.typeDependencies = ImmutableMap.copyOf(requireNonNull(typeDependencies, "typeDependencies is null"));
-        this.functionDependencies = ImmutableSet.copyOf(requireNonNull(functionDependencies, "functionDependencies is null"));
+        requireNonNull(signature, "signature is null");
+        requireNonNull(catalogHandle, "catalogHandle is null");
+        requireNonNull(functionId, "functionId is null");
+        requireNonNull(functionKind, "functionKind is null");
+        requireNonNull(functionNullability, "functionNullability is null");
+        typeDependencies = ImmutableMap.copyOf(requireNonNull(typeDependencies, "typeDependencies is null"));
```

Comment:
null check no needed. ImmutableMap.copyOf will take care of that.

Reply:
i would keep it, so that eventual NPE message indicates which record constructor parameter was null

Reply:
rnn check was already there so I'll leave it as it is